### PR TITLE
feat(ae): Add support for Technical API

### PR DESCRIPTION
**Pull request by brasseld**

### CODE REVIEW

File: ae/values.yaml [hunk 1]
```diff
@@ -186,4 +186,38 @@ engine:
...
+  services:
+    core:
+      http:
+        enabled: true
+        port: 18072
+        host: localhost
+        authentication:
+          type: basic
+          password: adminadmin
+        secured: false
```
Comment (kamiiiel) [inline ae/values.yaml:200]: Is this used anywhere?

File: ae/values.yaml [hunk 2]
```diff
@@ -186,4 +186,38 @@ engine:
...
+        ssl:
+          keystore:
+            type: "PKCS12"
+            path: "/p12/keystore"
```
Comment (kamiiiel) [inline ae/values.yaml:204]: Is this used anywhere?

Comment (kamiiiel) [inline ae/values.yaml:204]: Is this used anywhere?

File: ae/templates/engine-deployment.yaml [hunk 3]
```diff
@@ -102,6 +102,10 @@ spec:
           ports:
             - name: {{ .Values.engine.service.internalPortName }}
               containerPort: {{ .Values.engine.service.internalPort }}
+            {{- if .Values.engine.services.core.http.enabled }}
+            - name: {{ .Values.engine.name }}-techapi
+              containerPort: {{ .Values.engine.services.core.http.port | default "18072" }}
```
[the remark this answers is not in the document]
Comment (brasseld) [reply ae/templates/engine-deployment.yaml:107]: Yes, was done for keeping coherency with other products

File: ae/values.yaml (hunk 2 above)
Comment (brasseld) [reply ae/values.yaml:204]: Copy and paste :) will remove it

File: ae/templates/engine-service.yaml [hunk 4]
```diff
@@ -24,6 +24,12 @@ spec:
     - port: 5701
       name: {{ .Values.engine.name }}-hz
       protocol: TCP
+    {{- if .Values.engine.services.core.service.enabled }}
+    - port: {{ .Values.engine.services.core.service.externalPort }}
+      targetPort: {{ .Values.engine.services.core.http.port }}
+      protocol: TCP
+      name: {{ .Values.engine.name }}-techapi
+    {{- end }}
```
Comment (jhaeyaert) [inline ae/templates/engine-service.yaml:32]: I don't get this part 🧐
What is does mean `.Values.engine.services.core.service.enabled` ? Which "service" is enabled here ? Should it be `.Values.engine.services.core.service.http.enabled` instead ?

Same for `.Values.engine.services.core.service.externalPort` ?

Comment (jhaeyaert) [reply ae/templates/engine-service.yaml:32]: Ok it refers to the "core" service which corresponds to the technical api itself. The "http" is only a way to expose the technical api (it is the only supported for now). So makes sense to keep it like this and keep consistency with other products (apim, am). 👍

**Review by jhaeyaert — APPROVED** [review]

**Pull request merged**